src/css.js

```diff
@@ -257,7 +257,9 @@ jQuery.extend( {
 			if ( !hooks || !( "set" in hooks ) ||
 				( value = hooks.set( elem, value, extra ) ) !== undefined ) {
 
-				style[ name ] = value;
+				if ( support.styleWhitespace || !/^\s+$/.test( value ) ) {
+					style[ name ] = value;
+				}
 			}
```

Comment:
Is there harm in just punting on whitespace only string across all enviros without a need to add a support test?
You can simplify the check to something like `if (value && value == 0) {`. It takes advantage of the fact that
`"       " == 0` is `true`.

[7.1.3.1ToNumber Applied to the String Type](http://www.ecma-international.org/ecma-262/7.0/#sec-tonumber-applied-to-the-string-type)
> A StringNumericLiteral that is empty or contains only white space is converted to +0.